packages/react/src/components/DatePicker/DatePicker.base.tsx

```diff
@@ -377,6 +377,7 @@ export const DatePickerBase: React.FunctionComponent<IDatePickerProps> = React.f
           setSelectedDate(newlySelectedDate);
         }
       }
+      textFieldRef.current?.focus?.();
```

Comment:
In theory this should also cause `ITextFieldState.isFocused` to be set to true. Was that not happening?

Reply:
Nope, isFocused remained false

Reply:
That's really strange... `textFieldRef.current.focus()` calls `focus()` on the input element, which I assume should call TextField's `_onFocus` which updates the state. But in the debugger I see that's not happening (at least while debugging, which can mess up focus stuff). @dzearing or @khmakoto do you happen to have any ideas why this wouldn't work? 

Sorry for pushing on this so much, but if possible I'd like to either figure out why the current approach isn't working as expected, or just switch to always checking `document.activeElement` (but I'd also want to check with others to make sure that approach would work as expected). Using two different ways to determine whether the field has focus is too messy and seems likely to create future bugs.

Reply:
`textFieldRef.current.focus()` actually calls the public `focus()` method in TextField rather than the `_onFocus` method. I'm would also be interested to hear other's input on the matter but one thing I could do is set the state of `isFocused` to true in the `focus()` method which allows me to remove the `hasFocus` variable I created. 
https://github.com/microsoft/fluentui/blob/a2dd7d1c0789bc7ade44228caa2bb0a629dbe7bd/packages/react/src/components/TextField/TextField.base.tsx#L281-L285

Reply:
Right, but calling `this._textElement.current.focus()` seems like it ought to trigger the input's `onFocus` handler, which in this case is `_onFocus`. Unless I'm misunderstanding how that's supposed to work. (If so that seems like an important "gotcha" to know about.)

Reply:
Something is definitely weird here. I can observe the text input getting focused in browser dev tools, but yeah, the TextField focus handler isn't getting called. Tabbing again afterwards triggers the blur handler though 🤔

Reply:
This is somehow related to focus being called in the `useLayoutEffect` cleanup function. The `element.focus()` called there works as far as the DOM is concerned, but React's synthetic focus event appears to just not get called.

You can see this in Popup.tsx: https://github.com/microsoft/fluentui/blob/master/packages/react/src/components/Popup/Popup.tsx#L52-L82

If you wrap `originalElement.focus?.();` in a `setTimeout` on line 59, it works (and introduces some other component bugs, but the React focus event does fire).

Reply:
(that wasn't a suggestion on my part to wrap it in a setTimeout, just a way to observe the issue 😅)

Reply:
Thanks for the investigation! That's pretty concerning... We should definitely do some more searching online and run this by a few people to see if anyone is familiar with the issue. 

I'm also curious how this behaves with React 17 since it made some changes to how focusing works under the hood, and when I was trying to run TextField tests against React 17 some of them broke in weird ways (which I'd assumed was a test-only bug, but maybe not).

Reply:
Found the root cause, finally 😅 https://github.com/facebook/react/issues/17894#issuecomment-656094405

Reply:
@ecraig12345 how would you feel about using this PR to just use `:focus-within` for focus styling and remove all default styles from `.focused`, then open a new issue about the DOM focus/React onFocus/`useLayoutEffect` cleanup problem?

Reply:
@smhigley Thanks so much for digging into this and finding the root cause! Agreed we should make a separate issue to discuss (and possibly check for other instances of this within our codebase).

The problem with not updating `isFocused` state is that it breaks consumer expectations for how they can do styling overrides, since `ITextFieldStyleProps.focused` is a public API. So I'm not quite sure what the best path forward is.